### DISCUSSION TITLE
Connection string from appsettings.json

### DISCUSSION THREAD

**Discussion by alexanderbright** [fluentmigrator/fluentmigrator#1303] [Q&A] [answered]
Could you tell me if it is possible to read a connection string from appsettings.json when I run migrations with the tool instead of reading the connection string from command line arguments?

```
dotnet fm migrate ...
```

**Comment by jzabroski** [comment, marked as answer]
It's not a good idea to save connection strings in the appSettings.json for a CLI tool, because if you blow away the manifest, you will blow away your appSettings.json.  That said, appSettings.json is just a json document, so you might be able to workaround this by loading the appSettings.json file via PowerShell, capturing the connection string, and passing that to dotnet fm migrate.  Personally, I configure my deployments using Octopus Deploy so that I have a global repository of all deployment data.

Another option would be to create your own in-process runner.  We are working on #1208 to make this easier to accomodate.
